asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
Oh, I must have misunderstood that part.  I thought the per-request auth header was still being set inside the asana package, which would have access to the non-exported methods.  If those auth headers are being set by the caller of this package then yes you're right, they'd need to be exported.